substrate/frame/balances/src/lib.rs

```diff
@@ -911,8 +911,8 @@ pub mod pallet {
 			let result = T::AccountStore::try_mutate_exists(who, |maybe_account| {
 				let is_new = maybe_account.is_none();
 				let mut account = maybe_account.take().unwrap_or_default();
-				let did_provide =
-					account.free >= Self::ed() && Self::have_providers_or_no_zero_ed(who);
+				let did_provide = (account.reserved + account.free) >= Self::ed() &&
```

Comment:
```suggestion
				let did_provide = account.free >= Self::ed() &&
```

Rolling back this is the wrong fix.
I believe, that we need to fix that in the ensure_upgraded function.

If an account that is being upgraded does not provide for itself (free < ed), we should move ed from reserved to the free balance when we migrate the account, or decrement it's provider count, so that the logic here does not offset the provider count by 1.